src/main.c

```diff
@@ -591,7 +591,16 @@ init_tty(void)
 	if (use_alternate_screen)
 		tty_putp(enter_ca_mode);
 
-	tty_putp(clear_screen);
+	/* Emit enough lines to fit all choices. */
```

Comment:
It's pre-allocating the number of lines needed to output the query line and all
input lines that fits on the screen in order to determine the position of the
first line. This could probably be done by moving the cursor up the same amount
of times as the number of matching choices after printing the choices but this
requires more state and I prefer the single save and restore capability.

On Wed, Dec 16, 2015 at 12:29:01AM -0800, Calle Erlandsson wrote:
> > @@ -591,7 +591,16 @@ init_tty(void)
> >  	if (use_alternate_screen)
> >  		tty_putp(enter_ca_mode);
> >  
> > -	tty_putp(clear_screen);
> > +	/* Emit enough lines to fit all choices. */
> 
> What is the reason for emitting these lines?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/thoughtbot/pick/pull/79/files#r47747515